test/e2e/packetcapture_test.go

```diff
@@ -0,0 +1,728 @@
+						IP: &tcpServerPodIP,
+					},
+					CaptureConfig: crdv1alpha1.CaptureConfig{
+						FirstN: &crdv1alpha1.PacketCaptureFirstNConfig{
+							Number: 500,
+						},
+					},
+					FileServer: &crdv1alpha1.PacketCaptureFileServer{
+						URL: fmt.Sprintf("sftp://%s:30010/upload", controlPlaneNodeIPv4()),
+					},
+					Packet: &crdv1alpha1.Packet{
+						Protocol: &tcpProto,
+						IPFamily: v1.IPv4Protocol,
+						TransportHeader: crdv1alpha1.TransportHeader{
+							TCP: &crdv1alpha1.TCPHeader{
+								DstPort: &testNonExistPort,
+							},
+						},
+					},
+				},
+			},
+			expectedStatus: crdv1alpha1.PacketCaptureStatus{
+				Conditions: []crdv1alpha1.PacketCaptureCondition{
+					{
+						Type:               crdv1alpha1.PacketCaptureCompleted,
+						Status:             metav1.ConditionStatus(v1.ConditionTrue),
+						LastTransitionTime: metav1.Now(),
+						Reason:             "Timeout",
+						Message:            "context deadline exceeded",
+					},
+				},
+			},
+		},
+		{
+
+			name:      nonExistPodName,
+			ipVersion: 4,
+			srcPod:    pcToolboxPodName,
+			pc: &crdv1alpha1.PacketCapture{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: randName(fmt.Sprintf("%s-%s-", data.testNamespace, nonExistPodName)),
+				},
+				Spec: crdv1alpha1.PacketCaptureSpec{
+					Source: crdv1alpha1.Source{
+						Pod: &crdv1alpha1.PodReference{
+							Namespace: data.testNamespace,
+							Name:      pcToolboxPodName,
+						},
+					},
+					Destination: crdv1alpha1.Destination{
+						Pod: &crdv1alpha1.PodReference{
+							Namespace: data.testNamespace,
+							Name:      nonExistPodName,
+						},
+					},
+					CaptureConfig: crdv1alpha1.CaptureConfig{
+						FirstN: &crdv1alpha1.PacketCaptureFirstNConfig{
+							Number: 5,
+						},
+					},
+					FileServer: &crdv1alpha1.PacketCaptureFileServer{
+						URL: fmt.Sprintf("sftp://%s:30010/upload", controlPlaneNodeIPv4()),
+					},
+				},
+			},
+
+			expectedStatus: crdv1alpha1.PacketCaptureStatus{
+				Conditions: []crdv1alpha1.PacketCaptureCondition{
+					{
+						Type:               crdv1alpha1.PacketCaptureCompleted,
+						Status:             metav1.ConditionStatus(v1.ConditionFalse),
+						LastTransitionTime: metav1.Now(),
+						Reason:             "CaptureFailed",
+						Message:            fmt.Sprintf("failed to get Pod %s/%s: pods \"%s\" not found", data.testNamespace, nonExistPodName, nonExistPodName),
+					},
+				},
+			},
+		},
+	}
+	t.Run("testPacketCapture", func(t *testing.T) {
+		for _, tc := range testcases {
+			tc := tc
+			t.Run(tc.name, func(t *testing.T) {
+				t.Parallel()
+				runPacketCaptureTest(t, data, tc)
+			})
+		}
+	})
+}
+
+// testPacketCaptureTCP verifies if PacketCapture can capture tcp packets. this function only contains basic
+// cases with pod-to-pod.
+func testPacketCaptureBasic(t *testing.T, data *TestData) {
+	nodeIdx := 0
+	if len(clusterInfo.windowsNodes) != 0 {
+		nodeIdx = clusterInfo.windowsNodes[0]
+	}
```

Comment:
this is confusing to me because the PacketCapture implementation currently depends on BPF and should not work on Windows Nodes

Reply:
removed